libbeat/beat/template.go

```diff
@@ -0,0 +1,67 @@
+package beat
+
+/*
+
+Load the index template. This replaces the manual running of
+curl -XPUT 'http://localhost:9200/_template/filebeat?pretty' -d@/etc/filebeat/filebeat.template.json
+
+
+TODO:
+ - Cleanup TODO's
+ - Should the template also be automatically loaded by logstash-input-beats?
+
+*/
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+var template *string
+
+func init() {
+	template = flag.String("template", "", "Path the index template")
+}
+
+func (beat *Beat) loadTemplate() error {
+
+	// TODO: Fin a way to check if flag was even set. If it was set but no path, default path should be used.
+	if *template != "" {
+
+		output, ok := beat.Config.Output["elasticsearch"]
+		if !ok {
+			return fmt.Errorf("Elasticaserch output not enabled")
+		}
+
+		// TODO: Refactor elasticsearch output so existing client can be used
+		client := elasticsearch.NewClient("http://localhost:9200", "", nil, nil, "", "")
+		content, err := ioutil.ReadFile(*template)
+		if err != nil {
+			return fmt.Errorf("Could not load template: %s", err)
+		}
+		reader := bytes.NewReader(content)
+
+		// TODO: This should be retrieved from the client
+		indexName := output.Index
+		if indexName == "" {
+			indexName = beat.Name
+		}
+		logp.Debug("test", "Load template for index: %s", indexName)
+		status, resp, err := client.ExecRequest("PUT", "http://localhost:9200/_template/"+indexName, reader)
```

Comment:
what if template has been applied already? What if template is incompatible to actual index?